core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java

```diff
@@ -30,7 +30,9 @@
  */
 public class PrefixQueryParser implements QueryParser<PrefixQueryBuilder> {
 
-    private static final ParseField NAME_FIELD = new ParseField("_name").withAllDeprecated("query name is not supported in short version of prefix query");
+    public static final ParseField NAME_FIELD = new ParseField("_name").withAllDeprecated("query name is not supported in short version of prefix query");
```

Comment:
Took me a while to understand why this is here: so I think NAME_FIELD so far was used only in the short version of prefix query, which is: `{ "prefix" : { "user" : "ki" }}`, so by deprecating "_name" it would throw exception in STRICT mode. So by using AbstractQueryBuilder.NAME_FIELD in the allowed position and this one in the deprecated short version position this should be okay. Can you rename it to something like DEPRECATED_NAME_FIELD or something though? Maybe somebody else should look at this also.